t/t3907-stash-show-config.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/sh
+
+test_description='Test git stash show configuration.
+
+'
+
+. ./test-lib.sh
+
+config_stash_stat_and_patch () {
+	if test "$1" != "unset"
+	then
+		git config stash.showStat "$1"
+	fi
+
+	if test "$2" != "unset"
+	then
+		git config stash.showPatch "$2"
+	fi
+}
+
+unset_config () {
+	if git config --bool stash.showStat >/dev/null
+	then
+		git config --unset stash.showStat
+	fi
+
+	if git config --bool stash.showPatch >/dev/null
+	then
+		git config --unset stash.showPatch
+	fi
+}
+
+get_expected_output () {
+	FLAGS=
+
+	if test "$(git config --bool stash.showStat || echo true)" = "true"
+	then
+		FLAGS=--stat
+	fi
+
+	if test "$(git config --bool stash.showPatch || echo false)" = "true"
+	then
+		FLAGS=${FLAGS}${FLAGS:+ }-p
+	fi
+
+	git diff $FLAGS >expect
+}
+
+get_actual_output () {
+	git stash &&
+	git stash show >actual
+}
+
+check_output () {
+	get_expected_output &&
+	get_actual_output &&
+	test_cmp expect actual &&
+	rm expect actual
+}
+
+options="unset true false"
+
+test_expect_success 'setup' '
+	echo 1 >file &&
+	git add file &&
+	git commit -m initial
+'
+
+for i in $options
+do
+	for j in $options
+	do
+		if test "$i" = "false" && (test "$j" = "false" || test "$j" = "unset")
+		then
+			test_expect_success "showStat $i and showPatch $j" '
+				config_stash_stat_and_patch $i $j &&
+				echo 2 >file &&
+				get_actual_output &&
+				test_must_be_empty actual &&
+				rm actual
+			'
+			continue
+		fi
+
+		unset_config
+		test_expect_success "showStat $i and showPatch $j" '
+			config_stash_stat_and_patch $i $j &&
+			echo 2 >file &&
+			check_output
+		'
+	done
+done
```

Comment:
It is tempting to do it in a loop, as it is repetitive code.

However, I would suggest to refrain from doing that as it makes debugging regressions a pain in the back. I'd rather define a couple nice shell functions, and then spell out each invocation individually.